includes/polling/os/windows.inc.php

```diff
@@ -8,179 +8,116 @@
 // sysDescr.0 = STRING: Hardware: AMD64 Family 16 Model 8 Stepping 0 AT/AT COMPATIBLE - Software: Windows Version 6.1 (Build 7600 Multiprocessor Free)
-if (strstr($poll_device['sysDescr'], 'x86')) {
-    $hardware = 'Generic x86';
-}
-
-if (strstr($poll_device['sysDescr'], 'ia64')) {
-    $hardware = 'Intel Itanium IA64';
-}
 
-if (strstr($poll_device['sysDescr'], 'EM64')) {
-    $hardware = 'Intel x64';
-}
-
-if (strstr($poll_device['sysDescr'], 'AMD64')) {
+if (str_contains($poll_device['sysDescr'], 'AMD64')) {
     $hardware = 'AMD x64';
-}
-
-if (strstr($poll_device['sysDescr'], 'Intel64')) {
+} elseif (str_contains($poll_device['sysDescr'], array('EM64', 'Intel64'))) {
     $hardware = 'Intel x64';
+} elseif (str_contains($poll_device['sysDescr'], 'x86')) {
+    $hardware = 'Generic x86';
+} elseif (str_contains($poll_device['sysDescr'], 'ia64')) {
+    $hardware = 'Intel Itanium IA64';
 }
 
 if ($poll_device['sysObjectID'] == 'enterprises.311.1.1.3.1.1') {
-    if (strstr($poll_device['sysDescr'], 'Build Number: 1057')) {
-        $version = 'NT 3.51 Workstation';
-    }
-
-    if (strstr($poll_device['sysDescr'], 'Build Number: 1381')) {
-        $version = 'NT 4.0 Workstation';
-    }
-
-    if (strstr($poll_device['sysDescr'], 'Build 2195')) {
-        $version = '2000 (NT 5.0)';
-    }
-
-    if (strstr($poll_device['sysDescr'], 'Build 2600')) {
-        $version = 'XP (NT 5.1)';
-    }
-
-    if (strstr($poll_device['sysDescr'], 'Build 3790')) {
-        $version = 'XP x64 (NT 5.2)';
-    }
-
-    if (strstr($poll_device['sysDescr'], 'Build 6000')) {
-        $version = 'Vista (NT 6.0)';
-    }
-
-    if (strstr($poll_device['sysDescr'], 'Build 6001')) {
-        $version = 'Vista SP1 (NT 6.0)';
-    }
-
-    if (strstr($poll_device['sysDescr'], 'Build 6002')) {
-        $version = 'Vista SP2 (NT 6.0)';
-    }
-
-    if (strstr($poll_device['sysDescr'], 'Build 7600')) {
-        $version = '7 (NT 6.1)';
-    }
-
-    if (strstr($poll_device['sysDescr'], 'Build 7601')) {
-        $version = '7 SP1 (NT 6.1)';
-    }
-
-    if (strstr($poll_device['sysDescr'], 'Build 9200')) {
-        $version = '8 (NT 6.2)';
-    }
-
-    if (strstr($poll_device['sysDescr'], 'Version 6.3 (Build 9200')) {
-        $version = '8.1 (NT 6.3)';
-    }
-
-    if (strstr($poll_device['sysDescr'], 'Build 9600')) {
-        $version = '8.1 SP1 (NT 6.2)';
-    }
-
-    if (strstr($poll_device['sysDescr'], 'Version 6.3 (Build 10')) {
+    // Client
+    if (str_contains($poll_device['sysDescr'], 'Build 14393')) {
+        $version = '10 AU (NT 6.3)';
+    } elseif (str_contains($poll_device['sysDescr'], 'Build 10586')) {
+        $version = '10 U1 (NT 6.3)';
+    } elseif (str_contains($poll_device['sysDescr'], 'Build 10240')) {
         $version = '10 (NT 6.3)';
+    } elseif (str_contains($poll_device['sysDescr'], 'Build 9600')) {
+        $version = '8.1 U1 (NT 6.3)';
```

Comment:
Yes, there was no service pack 1 for 8.1, only "8.1 update 1"
The one I was unsure about was windows 10 "anniversery update" should it be U2 or AU?  Left it AU for now.